dask/bytes/core.py

```diff
@@ -112,13 +112,31 @@ def read_bytes(
                     "Backing filesystem couldn't determine file size, cannot "
                     "do chunked reads. To read, set blocksize=None."
                 )
-            off = list(range(0, size, blocksize))
-            length = [blocksize] * len(off)
-            if not_zero:
-                off[0] = 1
-                length[0] -= 1
-            offsets.append(off)
-            lengths.append(length)
+
+            if size:
+                # shrink blocksize to give same number of parts
+                if size % blocksize and size > blocksize:
+                    blocksize = size / (size // blocksize)
+                place = 0
+                off = [0]
+                length = []
+
+                # figure out offsets, spreading around spare bytes
+                while size - place > (blocksize * 2) - 1:
+                    place += blocksize
+                    off.append(int(place))
+                    length.append(off[-1] - off[-2])
+                length.append(size - off[-1])
+
+                if not_zero:
+                    off[0] = 1
+                    length[0] -= 1
+                offsets.append(off)
+                lengths.append(length)
+            else:
```

Comment:
I think this clause only gets activated when `size=0`, is that right? At first I was a bit confused about how this clause would get triggered, given the `if size is None` check above. If the number zero is the only other case which would return False, then I think we need to put all three conditions into the same if then else statement.

Reply:
```python
if size is None:
    raise ValueError(
        ...
elif size == 0:
    # skip empty
    ...
else:
   # shrink blocksize to give same number of parts
    ...
```

Reply:
Yes, your understanding is correct, I'll move blocks around for clarity.